### PR TITLE
chore: remove phf from static files

### LINKED ISSUE #9946: Remove PHF from Static Files
### Describe the feature

Fixes https://github.com/paradigmxyz/reth/issues/9939

## Problem
We don't use perfect hash functions for any static file segments, but they cause dependency bloat and compilation errors for Wasm.

## Solution

Remove `ph` dependency from `reth-nippy-jar` crate without breaking backward compatibility of configuration files.

This will require leaving a dummy `phf` field for compatibility reasons https://github.com/paradigmxyz/reth/blob/750eb26256b6fcea0ad378a8cf08fe28f2dc0a7b/crates/storage/nippy-jar/src/lib.rs#L115-L116

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by nkysg**
Closes https://github.com/paradigmxyz/reth/issues/9946

use empty enum as dummy Functions

### CODE REVIEW

**Review by shekhirin — CHANGES_REQUESTED** [review]
We can also remove this https://github.com/paradigmxyz/reth/blob/97e153ab64c807fcc9860d37368e43a51f8f058b/crates/storage/nippy-jar/src/lib.rs#L396-L400, so it will not require us to have `set_keys` implemented anymore, and we can have just an empty enum or any other dummy field even without a hashmap.

File: crates/storage/nippy-jar/src/lib.rs [hunk 1]
```diff
@@ -376,7 +337,7 @@ impl<H: NippyJarHeader> NippyJar<H> {
     /// later on inserted.
     ///
     /// Currently collecting all items before acting on them.
-    pub fn prepare_index<T: PHFKey>(
+    pub fn prepare_index<T: AsRef<[u8]> + Sync + Clone>(
```
Comment (shekhirin) [inline crates/storage/nippy-jar/src/lib.rs:340]: we can remove this method altogether, we only use it in tests

**Review by shekhirin — CHANGES_REQUESTED** [review]
almost there, also @joshieDo PTAL

**Review by shekhirin — APPROVED** [review]
Amazing, LGTM! Waiting for @joshieDo review, and we will also need to check that this doesn't break existing clients.

File: crates/storage/nippy-jar/src/cursor.rs [hunk 2]
```diff
@@ -139,29 +108,10 @@ impl<'a, H: NippyJarHeader> NippyJarCursor<'a, H> {
...
-                        .expect("built from same set") as u64;
-                    return self.next_row_with_cols(mask)
-                }
-            }
-        } else {
-            return Err(NippyJarError::UnsupportedFilterQuery)
-        }
-
-        Ok(None)
+        Err(NippyJarError::UnsupportedFilterQuery)
```
Comment (mattsse) [inline crates/storage/nippy-jar/src/cursor.rs:114]: why can't we remove this function if this now always returns an error, this is also not a trait function, so unclear why removing this is sound

**Review by mattsse — COMMENTED** [review]
ptal @joshieDo

File: crates/storage/nippy-jar/src/cursor.rs (hunk 2 above)
Comment (shekhirin) [reply crates/storage/nippy-jar/src/cursor.rs:114]: fair, we can remove it completely

**Comment by emhane** [comment]
@joshieDo status on testing?

**Review by joshieDo — APPROVED** [review]
lgtm, pushed `test_config_serialization`. we can also remove the filter in a follow-up

would merge after release though

**Comment by emhane** [comment]
> lgtm, pushed `test_config_serialization`. we can also remove the filter in a follow-up
> 
> would merge after release though

would you open an issue pls @joshieDo ? or @nkysg if you have enough context?

**Review by mattsse — APPROVED** [review]

**Pull request merged**